modules/ROOT/partials/depl-examples/ubuntu-compose/shared-setup.adoc

```diff
@@ -573,16 +575,91 @@ docker volume ls
+
+* Each upgrade consists of a set of commands:
++
+[source,bash]
+----
+sudo docker compose stop \
+sudo docker compose pull \
+sudo docker compose up -d --remove-orphans
+----
+
+* If there is no release gap, run the command block once.
+
+* For *any* https://owncloud.dev/ocis/release_roadmap/#dates[release gap], you must run the command block from above once and set the respective release in the `OCIS_DOCKER_TAG` _before_ pulling. Use no value for the last release, it defaults to `latest`.
+
+* Check if there are any Infinite Scale config changes. To do so, run a `ocis init --diff` plus apply any patches. The detailed how-to is described in the xref:migration/upgrading-ocis.adoc[Upgrading Infinite Scale] guide for 7.0.0. When done, bring the deployment up.
+
+* Finally, you can remove any old images if no longer used.
+
+Note that we advice to manually check if the deployment source has changed. If this is the case, stop the deployment, backup your existing deployment source/configuration and update the xref:download-and-transfer-the-example[deployment sources], then re-apply your configuration settings and bring the deployment up.
+
 endif::[]
 
-If new Infinite Scale releases are available, you *must not* skip any version in between the current running and the latest available release for internal upgrade reasons. All versions need to be downloaded and started one time. For more details see the https://owncloud.dev/ocis/release_roadmap/#updating-and-overlap[Updating and Overlap] description in the developer documentation.
+ifeval::["{version-type}" != "rolling"]
+
+Follow these steps to update between production releases:
+
+For patch releases only!::
++
+--
+These commands will stop the deployment, updates all images and starts the deployment again. This works because the relevant `xxx_TAG` variables in the `.env` file are set to `latest` by default while others have hardcoded versions set.
+
+[source,bash]
+----
+sudo docker compose stop \
+sudo docker compose pull \
+sudo docker compose up -d --remove-orphans
+----
+{empty}
+--
 
-* If there is no release gap, you can update by just stopping and starting the compose environment.
-* For *any* https://owncloud.dev/ocis/release_roadmap/#dates[release gap], you must stop the compose environment with `docker compose down`, set the `OCIS_DOCKER_TAG` variable in the `.env` file accordingly, pull the new image with `docker compose pull` and start the compose environment with `docker compose up -d` again. For the last release, you can remove any setting of `OCIS_DOCKER_TAG` as `latest` is used by default.
+endif::[]
+
+=== Upgrading
+
+For all major and minor releases including from rolling to production::
```

Comment:
rolling is often ahead of production.
I suppose that an upgrade from rolling to production is only allowed just when a brand-new production release happens (e.g. upgrade from some 6.*.* rolling release to 7.0.0)
Is that correct? If so, then do we mention that somewhere?

Reply:
Good point, I added "former" before rolling.